index.js

```diff
@@ -250,12 +250,16 @@ module.exports = class Hypercore extends EventEmitter {
     }
 
     if (this.opened) ensureEncryption(s, opts)
-
-    this.sessions.push(s)
+    this._addSession(s)
 
     return s
   }
 
+  _addSession (s) {
+    this.sessions.push(s)
+    if (this.core) this.core.active++
```

Comment:
I'm not seeing how `this.core` can be unset. The book-keeping code becomes much easier to reason about without that `if`. So if it's indeed unneeded, better to remove it

 `_addSession ` is called either from `_openFromExisting(...)`, or in `session(...)`

In `session(...)` it runs after `_passCapabilities`, which sets `this.core`

In `_openFromExisting` it also runs after `_passCapabilities